classes/template.php

```diff
@@ -330,8 +330,14 @@ public function generate_pdf(bool $preview = false, ?int $userid = null, bool $r
             if (empty($filename)) {
                 $filename = get_string('certificate', 'customcert');
             }
-
-            $filename = clean_filename($filename . '.pdf');
+            
+            $sql = "SELECT course,fullname
+                    FROM   {customcert} c, {course} cr
+                    WHERE  c.id = :templateid
+                    AND c.course = cr.id";
+            $course = $DB->get_record_sql($sql, array('templateid' => $this->id));
```

Comment:
The SQL parameters have to be aligned. See https://moodledev.io/general/development/policies/codingstyle/sql#indentation.